src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -423,7 +426,31 @@ class ContentSidebar extends PureComponent<Props, State> {
      * @return {void}
      */
     fetchVersionsSuccessCallback = (versions: FileVersions): void => {
-        this.setState({ versions, versionError: undefined });
+        const { entries } = versions;
```

Comment:
Move this code in the Versions.api file. So that we do larger scale data manipulation outside the component and the component just gets the data not knowing about what got shifted around. (This is how sorting works for content picker and explorer). If in the future we decide to cache this end result (so that we don't sort and map repeatedly), then its useful to have it there.

Also double check if we can re-use util/sorter.js maybe?
I find it weird/silly that the api doesn't already return sorted results. Probably double check that too.

Reply:
Is something like this not needed for comments/tasks as well?

Reply:
Yup. Added methods for that too